src/common/commonutils/CommonUtils.c

```diff
@@ -462,3 +462,218 @@ bool FileExists(const char* name)
 {
     return ((NULL != name) && (-1 != access(name, F_OK))) ? true : false;
 }
+
+bool ParseHttpProxyData(const char* proxyData, char** proxyHostAddress, int* proxyPort, char** proxyUsername, char** proxyPassword, void* log)
+{
+    // We accept the proxy data string to be in one of two following formats:
+    //
+    // "http://server:port"
+    // "http://username:password@server:port"
+    //
+    // ..where the prefix can be either lowercase "http" or uppercase "HTTP"
+
+    const char httpPrefix[] = "http://";
+    const char httpUppercasePrefix[] = "HTTP://";

```

Comment:
Is the prefix for `HTTPS_PROXY` also "http" / "HTTP" ?

Reply:
Read the comment. The HTTPS prefix is required. Also see the test and how HTTPS prefixed options are not allowed.